web/src/use-portal-connection.tsx

```diff
@@ -0,0 +1,35 @@
+import {
+  connectionDatabase,
+  connectionHost,
+  connectionPassword,
+  connectionUser,
+} from "@/data/recoil";
+import { useRecoilState } from "recoil";
+import { useLocation } from "react-router-dom";
+
+export function usePortalConnection() {
```

Comment:
Please remove this hook, and instead implement portal connections in a similar way to vaporConnectionConfig in recoil.ts.

For the query params please use:
* portalHostname
* portalCredentials

Once you have a working portalConnectionConfig atom you can simply read it in the connectionConfig atom.